docs/tutorials/tabular_prediction/tabular-indepth.md

```diff
@@ -137,68 +136,146 @@ To output predicted class probabilities instead of predicted classes, you can us
+
+
+## Accelerating inference
+
+We describe multiple ways to reduce the time it takes for AutoGluon to produce predictions.
 
-## Maximizing predictive performance
+### Keeping models in memory
 
-To get the best predictive accuracy with AutoGluon, you should generally use it like this:
+By default, AutoGluon loads models into memory one at a time and only when they are needed for prediction. This strategy is robust for large stacked/bagged ensembles, but leads to slower prediction times. If you plan to repeatedly make predictions (e.g. on new datapoints one at a time rather than one large test dataset), you can first specify that all models should be loaded into memory as follows:
 
 ```{.python .input}
-long_time = 60 # for quick demonstration only, you should set this to longest time you are willing to wait
-predictor = task.fit(train_data=train_data, label=label_column, eval_metric=metric, auto_stack=True, time_limits=long_time)
+predictor._learner.persist_trainer() # load models into memory (still experimental, may fail for big ensembles)
+
+num_test = 20
+preds = np.array(['']*num_test, dtype='object')
+for i in range(num_test):
+    datapoint = test_data_nolabel.iloc[[i]]
+    pred_numpy = predictor.predict(datapoint)
+    preds[i] = pred_numpy[0]
+
+perf = predictor.evaluate_predictions(y_test[:num_test], preds, auxiliary_metrics=True)
+print("Predictions: ", preds)
+
+predictor = task.load(output_directory) # reset predictor
 ```
 
-This command implements the following strategy to maximize accuracy:
+### Collapsing bagged ensembles via refit_full
 
-- Specify the `auto_stack` argument, which allows AutoGluon to automatically construct model ensembles based on multi-layer stack ensembling with repeated bagging, and will greatly improve the resulting predictions if granted sufficient training time.
+For a ensemble predictor trained with bagging (as done above), recall there ~10 bagged copies of each individual model trained on different train/validation folds. We can collapse this bag of ~10 models into a single model that's fit to the full dataset, which can greatly reduce its memory/latency requirements. Below we refit such models for every model-type but you can alternatively do this for just a particular model-type by specifying the `model` argument of `refit_full()`.
```

Comment:
Should mention that this often reduces model accuracy due to the lack of bagged averaging.